habitat_baselines/il/data/eqa_cnn_pretrain_data.py

```diff
@@ -0,0 +1,227 @@
+import os
+import random
+from typing import List
+
+import cv2
+import lmdb
+import numpy as np
+from torch.utils.data import Dataset
+from tqdm import tqdm
+
+import habitat
+from habitat import logger
+from habitat.core.simulator import ShortestPathPoint
+
+
+class EQACNNPretrainDataset(Dataset):
+    """Pytorch dataset for Embodied Q&A's feature-extractor"""
+
+    def __init__(self, config, mode="train"):
+        """
+        Args:
+            env (habitat.Env): Habitat environment
+            config: Config
+            mode: 'train'/'val'
+        """
+        self.config = config.TASK_CONFIG
+        self.mode = mode
+        self.dataset_path = config.DATASET_PATH
+
+        self.disk_cache_exists = self.check_cache_exists()
+
+        if not self.disk_cache_exists:
+            """
+            for each scene > load scene in memory > save frames for each
+            episode corresponding to each scene
+            """
+            try:
+                self.make_dataset_dirs()
+            except FileExistsError:
+                pass
+
+            logger.info("Saving rgb, seg, depth data to database.")
+
+            self.env = habitat.Env(config=self.config)
+            self.episodes = self.get_all_episodes(config)
+
+            self.scene_ids = []
+            self.scene_episode_dict = {}
+
+            # preparing a dict that stores list of episodes for each scene
+            for ep in self.episodes:
+                if ep.scene_id not in self.scene_ids:
+                    self.scene_ids.append(ep.scene_id)
+                    self.scene_episode_dict[ep.scene_id] = [ep.episode_id]
```

Comment:
Just add the episode instead of the episode ID to the dictionary.  That will simplify a lot of the logic.

Reply:
Done in bbb7a3a, thanks!